### PR TITLE
Fix LsaOpenPolicy interop definition

### LINKED ISSUE #36724: Suspicious definition of `LsaOpenPolicy` in Interop code
We have the following (formatted) definition in `Interop.LsaOpenPolicy`:

```cs
[DllImport(Interop.Libraries.Advapi32, EntryPoint = "LsaOpenPolicy", SetLastError = true, CharSet = CharSet.Unicode)]
internal static extern uint LsaOpenPolicy(
    string systemName,
    ref LSA_OBJECT_ATTRIBUTES attributes,
    int accessMask,
    out SafeLsaPolicyHandle handle
);
```

Notice that this takes a `string` as the first parameter.

In the documentation for `LsaOpenPolicy` found at https://docs.microsoft.com/en-us/windows/desktop/api/ntsecapi/nf-ntsecapi-lsaopenpolicy the definition is different:
```c
NTSTATUS LsaOpenPolicy(
  PLSA_UNICODE_STRING    SystemName,
  PLSA_OBJECT_ATTRIBUTES ObjectAttributes,
  ACCESS_MASK            DesiredAccess,
  PLSA_HANDLE            PolicyHandle
);
```

Note that the first parameter is a pointer to `LSA_UNICODE_STRING` defined in the documentation (https://docs.microsoft.com/en-gb/windows/desktop/api/lsalookup/ns-lsalookup-_lsa_unicode_string) as:
```c
typedef struct _LSA_UNICODE_STRING {
  USHORT Length;
  USHORT MaximumLength;
  PWSTR  Buffer;
} LSA_UNICODE_STRING, *PLSA_UNICODE_STRING;
```

Surely marshalling a `string` would not be the same as marshalling an `LSA_UNICODE_STRING`?

### PR DESCRIPTION
**Pull request by hughbe**
We have the following (formatted) definition in `Interop.LsaOpenPolicy`:

```cs
[DllImport(Interop.Libraries.Advapi32, EntryPoint = "LsaOpenPolicy", SetLastError = true, CharSet = CharSet.Unicode)]
internal static extern uint LsaOpenPolicy(
    string systemName,
    ref LSA_OBJECT_ATTRIBUTES attributes,
    int accessMask,
    out SafeLsaPolicyHandle handle
);
```

Notice that this takes a `string` as the first parameter.

In the documentation for `LsaOpenPolicy` found at https://docs.microsoft.com/en-us/windows/desktop/api/ntsecapi/nf-ntsecapi-lsaopenpolicy the definition is different:
```c
NTSTATUS LsaOpenPolicy(
  PLSA_UNICODE_STRING    SystemName,
  PLSA_OBJECT_ATTRIBUTES ObjectAttributes,
  ACCESS_MASK            DesiredAccess,
  PLSA_HANDLE            PolicyHandle
);
```

Note that the first parameter is a pointer to `LSA_UNICODE_STRING` defined in the documentation (https://docs.microsoft.com/en-gb/windows/desktop/api/lsalookup/ns-lsalookup-_lsa_unicode_string) as:
```c
typedef struct _LSA_UNICODE_STRING {
  USHORT Length;
  USHORT MaximumLength;
  PWSTR  Buffer;
} LSA_UNICODE_STRING, *PLSA_UNICODE_STRING;
```

Marshalling a `string` would not be the same as marshalling an `LSA_UNICODE_STRING`.

I believe the reason this was never encountered as a problem is because all callers currently pass `null` to this method. However, as part of consolidating interop code (especially r.e. System.DirectoryServices.*) this would be used with a non-null `systemName` 

Fixes #36724

### CODE REVIEW

**Review by filipnavara — APPROVED** [review]

File: src/Common/src/Interop/Windows/Advapi32/Interop.LsaOpenPolicy.cs [hunk 1]
```diff
@@ -11,6 +11,34 @@ internal static partial class Interop
     internal static partial class Advapi32
     {
         [DllImport(Interop.Libraries.Advapi32, EntryPoint = "LsaOpenPolicy", SetLastError = true, CharSet = CharSet.Unicode)]
-        internal static extern uint LsaOpenPolicy(string systemName, ref LSA_OBJECT_ATTRIBUTES attributes, int accessMask, out SafeLsaPolicyHandle handle);
+        private static extern uint LsaOpenPolicy(
+            ref SspiCli.LSA_UNICODE_STRING SystemName,
```
Comment (bartonjs) [inline src/Common/src/Interop/Windows/Advapi32/Interop.LsaOpenPolicy.cs]: What does the type have to do with SSPI?  It seems just generically Advapi32 to me.

Comment (filipnavara) [reply src/Common/src/Interop/Windows/Advapi32/Interop.LsaOpenPolicy.cs]: Strictly speaking it's actually NT kernel type (UNICODE_STRING) so NTDLL feels somewhat appropriate too.

Comment (hughbe) [reply src/Common/src/Interop/Windows/Advapi32/Interop.LsaOpenPolicy.cs]: We already have a `UNICODE_STRING` type I think

**Review by bartonjs — APPROVED** [review]

File: src/Common/src/Interop/Windows/Advapi32/Interop.LsaOpenPolicy.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/Common/src/Interop/Windows/Advapi32/Interop.LsaOpenPolicy.cs]: We should consider using the existing `UNICODE_STRING` and get rid of the other. It is actually defined that way internally:

``` C
NTSTATUS
LsaOpenPolicy(
    IN PUNICODE_STRING SystemName OPTIONAL,
    IN POBJECT_ATTRIBUTES ObjectAttributes,
    IN ACCESS_MASK DesiredAccess,
    IN OUT PLSA_HANDLE PolicyHandle
    )
```

There really is only one `UNICODE_STRING` in NT.

Comment (JeremyKuhne) [reply src/Common/src/Interop/Windows/Advapi32/Interop.LsaOpenPolicy.cs]: We're doing a `Mashal.Sizeof` for the object attributes in the calling code. We may just want to follow suit with what I did here: https://github.com/dotnet/corefx/blob/2f7844a8df4fef427df1a2bc77e9f34211a720a1/src/Common/src/Interop/Windows/NtDll/Interop.NtCreateFile.cs#L49

And use the `OBJECT_ATTRIBUTES`, killing the LSA version.

**Review by JeremyKuhne — APPROVED** [review]
Thanks for cleaning this up!

**Pull request merged**